pkg/minikube/cluster/cluster.go

```diff
@@ -122,12 +132,27 @@ func DeleteHost(api libmachine.API) error {
 	if err != nil {
 		return err
 	}
+
+	if err = unregister(host); err != nil {
+		return err
+	}
+
 	m := util.MultiError{}
 	m.Collect(host.Driver.Remove())
 	m.Collect(api.Remove(constants.MachineName))
 	return m.ToError()
 }
 
+// Un-register host VM
+func unregister(host *host.Host) error {
+	if drivers.MachineInState(host.Driver, state.Running)() {
```

Comment:
> This will silently go ahead deleting the VM w/o un-registrating it.

No it will no un-register VM silently because before unregister it need to check if vm is support registration and execute ssh command `cat /etc/os-release` so if a vm is any other state than running it will display error about vm status and above command is not executable on vm.